pkg/pillar/types/zedroutertypes.go

```diff
@@ -218,6 +219,48 @@ func (portConfig *DevicePortConfig) DoSanitize(
 	}
 }
 
+// CountMgmtPorts returns the number of management ports
+// Exclude any broken ones with Dhcp = DT_NONE
+func (portConfig *DevicePortConfig) CountMgmtPorts() int {
+
+	count := 0
+	for _, port := range portConfig.Ports {
+		if port.IsMgmt && port.Dhcp != DT_NONE {
+			count++
+		}
+	}
+	return count
+}
+
+// Equal compares two DevicePortConfig but skips things that are
+// more of status such as the timestamps and the ParseError
+// XXX Compare Version or not?
+// We compare the Ports in array order.
+func (portConfig *DevicePortConfig) Equal(portConfig2 *DevicePortConfig) bool {
+
+	if portConfig.Key != portConfig2.Key {
+		return false
+	}
+	if len(portConfig.Ports) != len(portConfig2.Ports) {
+		return false
+	}
+	for i, p1 := range portConfig.Ports {
+		p2 := portConfig2.Ports[i]
```

Comment:
Can't the order of ports change with same configuration per port?

Reply:
Sure, but we treat that as a change just like the old reflect.DeepEqual treated it as a change.